src/components/PopupWithForm.js

```diff
@@ -32,9 +32,26 @@ export default class PopupWithForm extends Popup {
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      this.close();
+      //this.close();
     });
     //The setEventListeners() method of the PopupWithForm class should add a submit event listener to the form
     //and call the setEventListeners() method of the parent class
   }
+
+  renderLoadingSave(loading) {
+    const submitButton = this._popupForm.querySelector(".modal__button");
+    if (loading) {
+      submitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
this._submitBtnText = this._submitBtn.textContent

is it supposed to be this._submitBtnText.textContent?